athena-sqlserver/src/main/java/com/amazonaws/athena/connectors/sqlserver/SqlServerMetadataHandler.java

```diff
@@ -253,13 +254,7 @@ public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest getTabl
 
                 // create a single split for view/non-partition table
                 if ("Y".equals(viewFlag) || rowCount == 0) {
-                    LOGGER.debug("Getting as single Partition: ");
-                    blockWriter.writeRows((Block block, int rowNum) ->
-                    {
-                        block.setValue(PARTITION_NUMBER, rowNum, ALL_PARTITIONS);
-                        //we wrote 1 row so we return 1
-                        return 1;
-                    });
+                    handleSinglePartition(blockWriter);
```

Comment:
I know this is a debt from previous change; can we however please change this truly ugly string flag to a boolean; 

instead of 

```
viewFlag = "VIEW".equalsIgnoreCase(resultSet.getString("TYPE_DESC")) ? "Y" : "N";
```

let's do this; 
```
viewFlag = "VIEW".equalsIgnoreCase(resultSet.getString("TYPE_DESC"));
```

Reply:
updated the code.

Reply:
thank you; really appreciate it.